libs/components/popovers/testing/src/popover/harness/popover-harness.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  ComponentHarness,
+  HarnessQuery,
+  TestElement,
+} from '@angular/cdk/testing';
+import { SkyOverlayHarness } from '@skyux/core/testing';
+
+/**
+ * Harness for interacting with an popover component in tests.
+ */
+export class SkyPopoverHarness extends ComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-popover-content';
+
+  #getBody = this.locatorFor('.sky-popover-body');
+  #getContainer = this.locatorFor('.sky-popover-container');
+  #getOverlay =
+    this.documentRootLocatorFactory().locatorForOptional(SkyOverlayHarness);
+  #getTitle = this.locatorForOptional('.sky-popover-title');
+
+  /**
+   * Returns a child harness.
+   */
+  public async queryHarness<T extends ComponentHarness>(
+    harness: HarnessQuery<T>
+  ): Promise<T | null> {
+    return this.locatorForOptional(harness)();
+  }
+
+  /**
+   * Returns child harnesses.
+   */
+  public async queryHarnesses<T extends ComponentHarness>(
+    harness: HarnessQuery<T>
+  ): Promise<T[]> {
+    return this.locatorForAll(harness)();
+  }
+
+  /**
+   * Returns a child test element.
+   */
+  public async querySelector(selector: string): Promise<TestElement | null> {
```

Comment:
If we're going to allow consumers to query any selector/harness in the popover, we need to run the query within the context of the `.sky-popover-body` selector (and not the host selector) since the consumer might use this method to query for the `.sky-popover-container` or `.sky-popover-body` elements themselves. I would create a new, internal `SkyPopoverBodyHarness` with a host selector of `.sky-popover-body`. This harness would have the querying methods on it, and the `SkyPopoverHarness` would call into it.
```
#getBody = this.locatorFor(SkyPopoverBodyHarness);

public async querySelector(selector: string): Promise<TestElement | null> {
  return (await this.#getBody()).querySelector(selector);
}
```

Reply:
Added